File: python/metest.py
import json

def loadTest(filename):
    f = open("../metest/"+filename)
    # returns JSON object as a dictionary
    metest = json.load(f)
    for i in metest['metest']:
        print(i)
    f.close()
    return metest

def metests(testfiles):
    tests = {}
    for file in testfiles:
        json = loadTest(file)
        goal = str(json['metest']['goal'])
        tests[goal] = json
    return tests

def results(Tests):
    for test in Tests:
        print("test= "+ test)
        print(Tests[test])
        if (Tests[test]['metest']['actual'] != Tests[test]['metest']['expect']):
            print("Test failure at: " + Tests[test] )
        
        #f = open(test + '.json', 'wb')
        #f.write(str(Tests[test]))

File: C++/metest.cpp
#include "metest.h"
#include <stdio.h>
#include <fstream>
#include <iostream> //must go 
//Metest::Metest(){}
//Metest::~Metest(){}

bool Metest::loadTest(const std::string& filename, std::string& metest)
{
    // read in the json file
	std::ifstream f(filename, std::ifstream::in);
    json js; //create unitiialized json object
    f >> js; // initialize json object with what was read from file
    std::cout << js["metest"] << std::endl; // prints json object to screen
    metest = js.dump();

    return true;
}

bool Metest::metests(const std::vector<std::string>&testFilePaths, json& out)
{
    json out;
    for (auto testFilename: testFilePaths){
        std::string testData;
        loadTest(testFilename, testData );  
        json j = json::parse(testData);
        std::string goal = j["metest"]["goal"];
        out[goal] = j;
    }

    return true;
}

bool Metest::results(json js)
{
    for (auto test: js){
        std::string testName = test.get<std::string>();
        std::cout << js[testName] << std::endl;
        if (js[testName]["metest"]["actual"] == js[testName]["metest"]["expect"]){
            std::cout << "Failure at " << js[testName] << std::endl;
        }
    }
    return true;
}